dso-l2/src/main/java/com/tc/objectserver/entity/MessageDecoder.java

```diff
@@ -0,0 +1,29 @@
+/*
+ *
+ *  The contents of this file are subject to the Terracotta Public License Version
+ *  2.0 (the "License"); You may not use this file except in compliance with the
+ *  License. You may obtain a copy of the License at
+ *
+ *  http://terracotta.org/legal/terracotta-public-license.
+ *
+ *  Software distributed under the License is distributed on an "AS IS" basis,
+ *  WITHOUT WARRANTY OF ANY KIND, either express or implied. See the License for
+ *  the specific language governing rights and limitations under the License.
+ *
+ *  The Covered Software is Terracotta Core.
+ *
+ *  The Initial Developer of the Covered Software is
+ *  Terracotta, Inc., a Software AG company
+ *
+ */
+package com.tc.objectserver.entity;
+
+import org.terracotta.entity.EntityMessage;
+import org.terracotta.entity.MessageCodecException;
+
+/**
+ *
+ */
+public interface MessageDecoder {
```

Comment:
I don't see what this level of indirection buys us or how it is being implemented.  It looks like it depends on an API change.